src/prefect/environments/storage/s3.py

```diff
@@ -0,0 +1,154 @@
+import io
+from typing import TYPE_CHECKING, Any, Dict
+
+
+import cloudpickle
+import pendulum
+from slugify import slugify
+
+import prefect
+from prefect.environments.storage import Storage
+
+if TYPE_CHECKING:
+    from prefect.core.flow import Flow
+
+
+class S3(Storage):
+    """
+    S3 storage class.  This class represents the Storage interface for Flows stored as
+    bytes in an S3 bucket.
+
+    This storage class optionally takes a `key` which will be the name of the Flow object
+    when stored in S3. If this key is not provided the Flow upload name will take the form
+    `slugified-flow-name/slugified-current-timestamp`.
+
+    Args:
+        - aws_access_key_id (str, optional): AWS access key id for connecting to S3.
+            Defaults to the value set in the environment variable
+            `AWS_ACCESS_KEY_ID` or `None`
+        - aws_secret_access_key (str, optional): AWS secret access key for connecting to S3.
+            Defaults to the value set in the environment variable
+            `AWS_SECRET_ACCESS_KEY` or `None`
+        - aws_session_token (str, optional): AWS session key for connecting to S3
+            Defaults to the value set in the environment variable
+            `AWS_SESSION_TOKEN` or `None`
+        - bucket (str, optional): the name of the S3 Bucket to store the Flow
+        - key (str, optional): a unique key to use for uploading this Flow to S3
+    """
+
+    def __init__(
+        self,
+        aws_access_key_id: str = None,
+        aws_secret_access_key: str = None,
+        aws_session_token: str = None,
+        bucket: str = None,
+        key: str = None,
+    ) -> None:
+        self.flows = dict()  # type: Dict[str, str]
+        self.bucket = bucket or ""
```

Comment:
Why do you use an empty string as a default value here?

Reply:
Had talked about this with @wagoodman. Going to make the bucket a required argument because this won't work without it